### ISSUE TITLE
Is it a good idea for consumers to replace clientid? Why don't you align with Stan？

### ISSUE THREAD

**Issue by guojianyu** [nats-io/nats.go#939]

**Comment by AlexCuse** [comment]
Why didn't stan align with nats might be a better question.  Is there a use case where you're especially missing the ClientID?

**Comment by AlexCuse** [comment]
FWIW I believe `Name` is more or less equivalent and you should be able to use:

`nats.Name(myClientID)`

**Comment by guojianyu** [comment]
You're right. nats.Name(myClientID) can replace clientid but the usage like kafka.  
good usage can be borrowed but  want to keep the nats features.
community thinks more about design.
